mongo/writeconcern/writeconcern.go

```diff
@@ -17,34 +21,126 @@ import (
 
 // ErrNegativeWTimeout indicates that a negative WTimeout was specified.
+//
+// Deprecated: ErrNegativeWTimeout will be removed in Go Driver 2.0.
 var ErrNegativeWTimeout = errors.New("write concern `wtimeout` field cannot be negative")
 
-// WriteConcern describes the level of acknowledgement requested from MongoDB for write operations
-// to a standalone mongod or to replica sets or to sharded clusters.
+// A WriteConcern defines a MongoDB read concern, which describes the level of acknowledgment
+// requested from MongoDB for write operations to a standalone mongod, to replica sets, or to
+// sharded clusters.
+//
+// For more information about MongoDB write concerns, see
+// https://www.mongodb.com/docs/manual/reference/write-concern/
 type WriteConcern struct {
-	w interface{}
-	j bool
+	// W requests acknowledgment that the write operation has propagated to a specified number of
+	// mongod instances or to mongod instances with specified tags. It sets the the "w" option
+	// in a MongoDB write concern.
+	//
+	// W must be a type string or int value.
+	//
+	// For more information about the "w" option, see
+	// https://www.mongodb.com/docs/manual/reference/write-concern/#w-option
+	W interface{}
+
+	// Journal requests acknowledgment from MongoDB that the write operation has been written to the
+	// on-disk journal. It sets the "j" option in a MongoDB write concern.
+	//
+	// For more information about the "j" option, see
+	// https://www.mongodb.com/docs/manual/reference/write-concern/#j-option
+	Journal *bool
+
+	// WTimeout specifies a time limit for the write concern. It sets the "wtimeout" option in a
+	// MongoDB write concern.
+	//
+	// It is only applicable for "w" values greater than 1. Using a WTimeout and setting Timeout on
+	// the Client at the same time will result in undefined behavior.
+	//
+	// For more information about the "wtimeout" option, see
+	// https://www.mongodb.com/docs/manual/reference/write-concern/#wtimeout
+	WTimeout time.Duration
+}
 
-	// NOTE(benjirewis): wTimeout will be deprecated in a future release. The more general Timeout
-	// option may be used in its place to control the amount of time that a single operation can run
-	// before returning an error. Using wTimeout and setting Timeout on the client will result in
-	// undefined behavior.
-	wTimeout time.Duration
+// Majority returns a WriteConcern that requests acknowledgment that write operations have been
+// durably committed to the calculated majority of the data-bearing voting members. If journaling is
+// enabled on mongod, nodes only acknowledge when the write operation has been written to the
+// on-disk journal.
+//
+// For more information about write concern "w: majority", see
+// https://www.mongodb.com/docs/manual/reference/write-concern/#mongodb-writeconcern-writeconcern.-majority-
+func Majority() *WriteConcern {
+	return &WriteConcern{W: "majority"}
```

Comment:
Do we need to make `"majority"` a constant?

Reply:
We can add it as a constant to remove the 2 places it's repeated in `writeconcern.go`. I don't think we should export that constant because we generally want people to use the `Majority` convenience function rather than the struct literal.